mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -405,12 +414,37 @@ class CompletionValueGenerator(
   )(using Context): String =
     val suffix = sym.typeParams match
       case Nil => ""
-      case tparams => tparams.map(_ => "_").mkString("[", ", ", "]")
+      case tparams => tparams.map(_ => "?").mkString("[", ", ", "]")
     val bind = if hasBind then "" else "_: "
     bind + name + suffix
 end CompletionValueGenerator
 
-object CaseExtractors:
+class MatchCaseExtractor(
+    pos: SourcePosition,
+    text: String,
+    completionPos: CompletionPos,
+):
+  object MatchExtractor:
+    def unapply(path: List[Tree]) =
+      path match
+        // foo mat@@
+        case (sel @ Select(qualifier, name)) :: _
+            if "match".startsWith(name.toString()) && text.charAt(
+              completionPos.start - 1
+            ) == ' ' =>
```

Comment:
In the first, main PR about match case completions for scala3 we came to conclusion to show match completions after `two.match` but not `two.m`, `two.ma` etc. because I think most users aren't using this syntax and having match completion there might be confusing (also, `match` completions are exclusive, maybe we should change that?)

Reply:
I think we can make match completions not exclusive, to enable things like `xxx.match@@` and completion `xxx.matches()`, but it's better to not show match completions in `xxx.mat@@` (only for `xxx.match@@`)

Reply:
Ok, thx for clarifying!